channels/sync_runner.go

```diff
@@ -260,11 +260,14 @@ func AccessNameToPrincipalName(accessPrincipalName string) (principalName string
 func ottoValueToStringArray(value otto.Value) []string {
 	nativeValue, _ := value.Export()
 
-	result := base.ValueToStringArray(nativeValue)
+	result, nonStrings := base.ValueToStringArray(nativeValue)
 
 	if result == nil && !value.IsNull() && !value.IsUndefined() {
 		base.Warnf("SyncRunner: Non-string, non-array passed to JS callback: %s", base.UD(value))
 	}
+	if nonStrings != nil {
```

Comment:
We should do the same value.IsNull/value.IsUndefined check here as above, or rearrange the code so we do that once, and then warn based on result/nonStrings inside.

The reason is that a channel mapper for a field that doesn't exist on a doc is now always going to warn:

![Screenshot 2020-02-28 at 11 48 41](https://user-images.githubusercontent.com/1525809/75546372-487fa680-5a20-11ea-8152-0e34c550cf7b.png)

Reply:
Restructured the code with value.IsNull/value.IsUndefined check.